test/lib/generate-mock-query.js

```diff
@@ -0,0 +1,49 @@
+const {
+  lib: {
+    index: { isString },
+  },
+} = require('@0ti.me/tiny-pfp');
+const {
+  _,
+  sinon: { spy },
+} = deps;
+
+module.exports = function() {
+  let responses = [];
+
+  this.bindQueryAndResponse = (query, response) => {
+    responses.push({ query, response });
+
+    return this;
```

Comment:
So `bindQueryAndResponse` just pushes the request and response to an array, and then returns a mockQuery object, including responses?

Reply:
Yeah! It's like a more complex sinon.stub() that I wrote myself to make sure that I'm only executing SQL that I'm intending to.

I think many people consider this behavior bad practice (extra code within your tests), but I have always found that belief odd. I think you can definitely run into problems doing it, but I tend to be extra cautious and careful when writing logic into tests.

Plus I haven't figured out a good way to use `sinon.stub()` to properly return values given specific arguments when the function can be called many different times and ways. It always does something unexpected for me.